### PR TITLE
feat(umd): adding umd build

### PR DESCRIPTION
**Pull request by weslleyaraujo**
as mentioned in #87

* added minimal webpack config based on [Redux Thunk](https://github.com/gaearon/redux-thunk/blob/master/webpack.config.babel.js)
* support UMD build
* added note on README

I followed the same pattern in the `Makefile`, let me know if something else can be improved 😅

### CODE REVIEW

**Comment by timche** [comment]
Thanks for your effort to build this feature! I will have a look at it asap :)

File: README.md [hunk 1]
```diff
@@ -143,6 +143,30 @@ addTodo(Observable.of(
...
+```
+
+We are also supporting UMD build:
+
+```js
+var ReduxActions = window.ReduxActions.default;
+```
+
+**once again `.default` is required.**
+
```
Comment (timche) [inline README.md]: Could you please move this section up to [here](https://github.com/acdlite/redux-actions/blame/master/README.md#L15)?

**Comment by weslleyaraujo** [comment]
@timche moved the note on UMD and also made some improvements. We actually don't need the `default` flag since the entry point doesn't export redux-actions as `export default..`.

**Comment by timche** [comment]
Thanks!

Btw, have you tested it?

**Comment by weslleyaraujo** [comment]
Yes, already using the `dist` in a small project.. so far so good :)

File: README.md [hunk 2]
```diff
@@ -13,6 +13,16 @@ npm install --save redux-actions
 import { createAction, handleAction, handleActions } from 'redux-actions';
 ```
 
+## Note
```
Comment (timche) [inline README.md]: `#### Note` please, so it is not that big ;)

File: README.md [hunk 3]
```diff
@@ -13,6 +13,16 @@ npm install --save redux-actions
...
+
+We are also supporting a UMD build:
```
Comment (timche) [inline README.md]: We should link to the UMD build, any idea where? https://npmcdn.com/?

Comment (weslleyaraujo) [inline README.md]: that looks really nice, another option would be [cndjs][https://cdnjs.com/]... I would go for both 😅
I will check that 👍

**Comment by timche** [comment]
Would you mind to add a link/links to the UMD to finalise this PR? 😅

**Comment by weslleyaraujo** [comment]
sorry the delay 😅

how about using https://github.com/cdnjs/cdnjs/issues ?
looks quite simple, just request the lib and we are good to go :)

**Comment by timche** [comment]
No problem :)

On npmcdn we can already access the library without requesting it: https://npmcdn.com/redux-actions@0.10.1

**Comment by weslleyaraujo** [comment]
thats cool, so I suppose it will work fine o/

File: README.md [hunk 4]
```diff
@@ -13,6 +13,16 @@ npm install --save redux-actions
 import { createAction, handleAction, handleActions } from 'redux-actions';
 ```
 
+### Note
+
+We are also supporting a UMD build:
```
Comment (timche) [inline README.md]: Can you please add the npmcdn link here?

`We are also supporting a [UMD build](https://npmcdn.com/redux-actions/dist/redux-actions.min.js)`

**Comment by timche** [comment]
Also your branch needs to be rebased after some changes on the master to resolve the conflicts. Otherwise I really want to merge this :)

**Comment by weslleyaraujo** [comment]
nice! will do that then 💪

**Comment by weslleyaraujo** [comment]
I am not that familiar on rebasing but I hope I did it properly 😅
Let me know if something else needs to be done

File: README.md [hunk 5]
```diff
@@ -13,6 +13,16 @@ npm install --save redux-actions
 import { createAction, handleAction, handleActions } from 'redux-actions';
 ```
 
+### Note
```
Comment (yangmillstheory) [inline README.md]: I think this still needs to be made smaller: https://github.com/acdlite/redux-actions/pull/91#discussion_r67833083

Either `####`, or just prefix the statement with a less intrusive **Note:**.

Another nit on English:

> We also support a UMD build

File: webpack.config.babel.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+const plugins = [
+  new webpack.optimize.OccurenceOrderPlugin(),
+  new webpack.DefinePlugin({
+    'process.env.NODE_ENV': JSON.stringify(NODE_ENV),
+  }),
+];
+
+const filename = `redux-actions${NODE_ENV === 'production' ? '.min' : ''}.js`;
+
+NODE_ENV === 'production'  && plugins.push(
```
Comment (yangmillstheory) [inline webpack.config.babel.js]: `NODE_ENV === 'production'` is being used twice. Not a big deal, but what do you think about DRYing it up with a variable

```js
const isProduction = NODE_ENV === 'production'
```

Also, there's an extra space before `&& plugins.push(`.

File: Makefile [hunk 7]
```diff
@@ -0,0 +1,25 @@
+BIN=node_modules/.bin
```
Comment (yangmillstheory) [inline Makefile]: Hm, this was removed in #107, this PR is adding it back.

Could you [sync your fork](https://help.github.com/articles/syncing-a-fork/), and try to use https://github.com/acdlite/redux-actions/blob/master/package.json#L7 to build the UMD?

File: package.json [hunk 8]
```diff
@@ -45,7 +46,11 @@
     "eslint-watch": "^2.1.13",
     "flux-standard-action": "^0.6.0",
     "mocha": "^2.2.5",
-    "rimraf": "^2.5.3"
+    "rimraf": "^2.5.3",
+    "flux-standard-action": "^0.6.0",
+    "lodash.isplainobject": "^3.2.0",
```
Comment (yangmillstheory) [inline package.json]: Is there a reason `lodash.isplainobject` is being added back?

See https://github.com/acdlite/redux-actions/pull/107#discussion_r70169615.

File: Makefile (hunk 7 above)
Comment (timche) [inline Makefile]: We need a module bundler to build the UMD file and `webpack` does that, so it's ok.

Just change the `scripts` in `package.json` to something like this:
```json
// ...
"scripts": {
  "build": "npm run clean && npm run build:commonjs && npm run build:umd && npm run build:umd:min",
  "build:commonjs": "babel src --out-dir lib --ignore *-test.js",
  "build:umd": "NODE_ENV=development webpack",
  "build:umd:min": "NODE_ENV=production webpack",
  // ...
}
// ...
```

File: README.md (hunk 5 above)
Comment (timche) [inline README.md]: 👍

File: package.json (hunk 8 above)
Comment (timche) [inline package.json]: I'd say this just came back accidentally like the `Makefile`.

File: webpack.config.babel.js (hunk 6 above)
Comment (timche) [inline webpack.config.babel.js]: Adds more safety 👍

File: webpack.config.babel.js [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+export default {
+  module: {
+    loaders: [
+      { test: /\.js$/, loaders: ['babel-loader'], exclude: /node_modules/ },
+    ],
+  },
+
+  entry: [
+    './src/index',
+  ],
```
Comment (timche) [inline webpack.config.babel.js]: Array annotation can be omitted. `entry` can also receive just the string.

**Comment by yangmillstheory** [comment]
ping @weslleyaraujo, would love to get this work in with the updates suggested above

**Pull request closed without merging**